test/test_ResponseHandler.py

```diff
@@ -339,3 +346,33 @@ def test_set_response(self, db, utils, max_commands, monkeypatch):
                         else:
                             assert part[key]["returned"] == j + 1
                 assert len(self.response_blobs[set_key]) == expected_blobs
+    @pytest.mark.parametrize("cpq", range(1, 3))
+    def test_varying_response_w_index(self, db, utils, cpq):
+        self.cleanDB()
+        persons = EntityWithResponseDataCSV(
+            "./input/persons.adb.csv", self.requests, self.responses)
+        loader = ParallelLoader(db)
+        loader.ingest(persons, batchsize=99,
+                      numthreads=31,
+                      stats=True)
+        assert loader.error_counter == 0
+
+        dist = persons.df.groupby(persons.df.age // 10 * 10).count()
+
+        self.requests = {}
+        self.responses = {}
+        self.response_blobs = {}
+        generator = QGPersonsIndex(self.requests, self.responses,
+                              cpq, self.response_blobs)
+        querier = ParallelQuery(db)
+        querier.query(generator, batchsize=100,
+                      numthreads=1,
+                      stats=True)
+        dist_by_ages = {}
+        # req and resp are dict mapping index number to query.
+        for reqk, respk in zip(self.requests.keys(), self.responses.keys()):
+            req=self.requests[reqk]
+            age_group = req[0]["FindEntity"]["constraints"]["age"][1]
+            # in QGPersonIndex, age_group[1] is created by taking index
+            # and multiplying it by cqp and 10.
+            assert reqk == ( 0 if age_group == 0 else age_group / cpq / 10 )
```

Comment:
Why is this not?
```suggestion
            assert reqk ==  age_group // cpq // 10 )
```

Reply:
for some reason I was thinking denominator could be zero